pyiron/atomistics/master/elastic.py

```diff
@@ -0,0 +1,271 @@
+
+    @property
+    def _number_of_measurements(self):
+        return max(
+            self.input['min_num_measurements'],
+            int(self.input['min_num_points']/max(len(self.input['rotations']), 1))
+        )
+
+    def validate_ready_to_run(self):
+        super().validate_ready_to_run()
+        if self.input['use_symmetry'] and len(self.input['rotations'])==0:
+            rotations = self.ref_job.structure.get_symmetry(use_elements=self.input['use_elements'])['rotations']
+            _, indices = np.unique(np.round(rotations, 6), axis=0, return_inverse=True)
+            rotations = rotations[np.unique(indices)]
+            self.input['rotations'] = rotations.tolist()
+        if len(self.input['strain_matrices'])==0:
+            eps_lst = np.random.random((int(self._number_of_measurements), 3, 3))-0.5
+            eps_lst *= self.input['max_strain']
+            eps_lst += np.einsum('nij->nji', eps_lst)
+            self.input['strain_matrices'] = eps_lst.tolist()
+
+    def list_structures(self):
+        if self.ref_job.structure is not None:
+            return [parameter[1] for parameter in self._job_generator.parameter_list]
+        else:
+            return []
+
+    def collect_output(self):
+        if self.ref_job.server.run_mode.interactive:
+            ham = self.project_hdf5.inspect(self.child_ids[0])
+            for key in ['energy_tot', 'pressures', 'volume']:
+                if key in ham["output/generic"].list_nodes():
+                    self._output[key.split('_')[0]] = ham["output/generic/{}".format(key)]
+                else:
+                    self._output[key.split('_')[0]] = []
+        else:
+            output_dict = defaultdict(list)
+            for job_id in self.child_ids:
+                ham = self.project_hdf5.inspect(job_id)
+                for key in ['energy_tot', 'energy_pot', 'pressures', 'volume']:
+                    if key in ham["output/generic"].list_nodes():
+                        output_dict[key.split('_')[0]].append(ham["output/generic/{}".format(key)][-1])
+                    else:
+                        output_dict[key.split('_')[0]] = []
+                output_dict['id'].append(job_id)
+            for k,v in output_dict.items():
+                self._output[k] = np.array(v)
+        elastic_tensor, score = calc_elastic_tensor(
+            strain = self.input['strain_matrices'],
+            stress = -np.array(self._output['pressures']),
+            rotations = self.input['rotations'],
+            energy = self._output['energy'],
+            volume = self._output['volume'],
+            return_score = True,
+        )
+        self._output['fit_score'] = score
+        for k,v in calc_elastic_constants(elastic_tensor).items():
+            self._output[k] = v
+        with self.project_hdf5.open("output") as hdf5_out:
+            for key, val in self._output.items():
+                hdf5_out[key] = val
+
+    def get_C_by_orientation(self, orientation):
```

Comment:
Same as `list_structures`.

Reply:
Actually, I'd hold off here. IMO this should not be a public method. We need to go back into the base classes and tweak some stuff. I'm pretty sure this only gets called internally by the job itself when it's run with status=collect.

Reply:
In fact this can only be used by the user - it's not used internally. Here's how you'd use it: If you have a box in which a dislocation is meant to be inserted, you might have an orientation like [[1, 1, 1], [-1, 0, 1], [1, -2, 1]]. In which case, you can either create a box from the beginning with this orientation, or you rotate the elastic tensor. Since creating a box with a weird orientation both increases the number of atoms (and hence the computation time) and decreases the number of symmetries phonopy finds, usually it makes more sense to rotate the elastic tensor. And those who are familiar with the anisotropic elasticity theory (I can't remember who did the work) might know that a full elastic tensor is required to parametrise the strain field around a dislocation for the given orientation. So this function totally makes sense for such a case.

Reply:
But it's true that the name is not quite comprehensible so I change it now.